### PR TITLE
UI: Added a switch to show soft deleted teams on team details page

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on adding a switch to show soft deleted teams on team details page.

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

https://user-images.githubusercontent.com/51777795/191962885-aad3446d-c9bf-49de-99e4-a48d2ed27e98.mov

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

**Comment by chirag-madlani** [comment]
@aniketkatkar97 UI for switch looks different to me. Can you match it with rest of them we have?

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx [hunk 1]
```diff
@@ -172,7 +175,7 @@ const TeamDetailsV1 = ({
         teamUserPagin,
         isGroupType,
         isOrganization,
-        searchTerm ? table.length : undefined
+        table.length
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:178]: Is this change intentional?

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/TeamsPage.tsx [hunk 2]
```diff
@@ -242,6 +215,34 @@ const TeamsPage = () => {
...
+        name: data.name,
+        displayName: data.displayName,
+        description: data.description,
+        teamType: data.teamType as TeamType,
+        parents: fqn ? [selectedTeam.id] : undefined,
+      };
+      const res = await createTeam(teamData);
+      if (res) {
+        const parent = fqn ? selectedTeam.fullyQualifiedName : undefined;
+        handleAddTeam(false);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/TeamsPage.tsx:234]: Add this to finally block, so if API fails or succeeds modal will get closed.

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:178]: Yes it is. Just showing the count of teams in table.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Comment by ShaileshParmar11** [comment]
Can you confirm if the one below is also taken care of in this PR? I can not see the delete action here.
[Permanent delete option after soft deleting the team is not available.#7664](https://github.com/open-metadata/OpenMetadata/issues/7664)

**Comment by aniketkatkar97** [comment]
> Can you confirm if the one below is also taken care of in this PR? I can not see the delete action here. [Permanent delete option after soft deleting the team is not available.#7664](https://github.com/open-metadata/OpenMetadata/issues/7664)

@ShaileshParmar11 So the issue over there was not actually about the delete action visibility. User was not able to see the team details page for soft deleted team. Throwing the shown error in the issue. I will confirm once again with user about that.
